Lib/tempfile.py

```diff
@@ -518,18 +542,32 @@ def __iter__(self):
         for line in self.file:
             yield line
 
+    def __del__(self):
+        # This is to delete the temporary file in case delete = True,
+        # delete_on_close = False and no context manager was used
+        if self.delete:
+            try:
+                _os.unlink(self.name)
+            # It is okay to ignore FileNotFoundError. The file may have
+            # been deleted already.
+            except FileNotFoundError:
+                pass
+
```

Comment:
The finalizer should simply call `self._cleanup()`. 

```suggestion
        self._cleanup()
```
Note that `_cleanup()` is guarded by `_cleanup_called`, so it only does the work the first time it's called.  Also, note that `_cleanup()` first closes the file before trying to delete it. This is required because `os.open()` without `O_TEMPORARY` does not share `DELETE` access.